src/components/calendar.rs

```diff
@@ -1,28 +1,125 @@
+                    };
+                    div class={"absolute top-[calc(" (tp) "%)] bottom-[calc(" (bp) "%)] h-auto w-full"} {
+                        div class={"h-full w-full rounded-lg overflow-y-scroll text-xs lg:text-sm color-red bg-[hsl(" (sec.crn % 360) ",50%,25%)] flex flex-col box-sizing " (border)} {
+                            div {
+                                (sec.subject_code) " " (sec.course_code)
+                            }
+                            div {
+                                (sec.sequence_code)
+                            }
+                            div {
+                                (sec.enrollment) "/" (sec.enrollment_capacity) " (enrolment)"
+                            }
+                            div {
+                                (sec.waitlist) "/" (sec.waitlist_capacity) " (waitlist)"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    )
+}
 
+fn render_day(day: Day, timeslots: &Vec<Time>, sections: &Vec<&Section>) -> Markup {
+    let earliest = timeslots.first().unwrap();
+    let latest = timeslots.last().unwrap();
+    debug!(?day, ?sections);
     html!(
-        div id="calendar" class="w-full h-full flex" {
-            div class="w-12 flex flex-col" {
-                div class="text-[0.5rem] border-b-2 box-content grow" { "time" }
-                @for time in &timeslots {
-                    div class="text-[0.25rem] border-b-2 box-content grow" { (time) }
+        div class="flex-1 flex flex-col" {
+            div class="text-[calc(1.5vh)] lg:text-sm shrink flex justify-center items-center" { (day.to_string().to_lowercase()) }
+            div class="relative flex flex-col grow gap-0.5 lg:gap-1" {
+                @for _ in timeslots {
+                    div class="h-auto grow bg-neutral-100 dark:bg-neutral-600" {  }
+                }
+                @for section in sections {
+                    (render_section_cards(earliest, latest, &section, day))
                 }
             }
-            @for day in &days {
-                div class="grow flex flex-col" {
-                    div class="text-[0.5rem] border-b-2 box-content grow flex justify-center items-center" { (day) }
-                    @for _ in &timeslots {
-                        div class="text-[0.25rem] border-b-2 box-content grow" { "." }
+        }
+    )
+}
+
+pub fn render(sections: &Vec<&Section>) -> Markup {
+    debug!(?sections);
+
+    let meeting_times: Vec<&MeetingTime> = sections.iter().flat_map(|s| &s.meeting_times).collect();
+    debug!(?meeting_times);
+
+    let earliest: Time = meeting_times
+        .iter()
+        .flat_map(|mt| mt.start_time)
+        .min()
+        .unwrap_or(jiff::civil::time(8, 30, 0, 0));
+
+    let latest = meeting_times
+        .iter()
+        .flat_map(|mt| mt.end_time)
+        .max()
+        .unwrap_or(jiff::civil::time(23, 20, 0, 0))
+        - 20.minutes();
+
```

Comment:
I found a bug when the time ends in 40, the section scrolls off the screen:

![image](https://github.com/user-attachments/assets/deb1bc8b-7811-48da-89e5-448637d36db5)

Reply:
is there a course where the time ends in :40? or anything other than :20 or :50?

Reply:
> I'm not really sure I understand why we're subtracting 20 minutes here, could you explain?

of course! we subtract 20 because the last time block we want to render is the one which ends 10 minutes after the last course ends, hence it starts 20 mins before the last course ends. i can picture many edge cases where this will look wonky so gonna rework this a bit.

notes to self:
- round up to nearest hour from last meeting end
- cover mobile view case where starts on x:30 and ends on y:00 (n.5 hour time range may look weird with only half as many hour blocks)